channel/manager_test.go

```diff
@@ -1233,17 +1233,77 @@ func Test_Instance_MSCBaseState(t *testing.T) {
 		})
 	}
 }
-
-func Test_Instance_SetVPCState(t *testing.T) {
+func Test_Instance_SetCurrentVPCState(t *testing.T) {
 	type args struct {
 		vpcState VPCStateSigned
 	}
 	tests := []struct {
 		name     string
 		instance *Instance
 		args     args
-		wantErr  bool
-		wantSet  bool
+	}{
+		{
+			name: "valid-state",
+			instance: &Instance{
+				selfID:      aliceID,
+				peerID:      bobID,
+				roleChannel: Sender,
+			},
+			args: args{
+				vpcState: VPCStateSigned{
+					VPCState: VPCState{
+						ID:              []byte("sample-id"),
+						Version:         big.NewInt(1),
+						BlockedSender:   big.NewInt(10),
+						BlockedReceiver: big.NewInt(20),
+					},
+					SignSender:   types.Hex2Bytes("14c5811728aed81bc8f7eb1fb71f45d7e7364561f88987da73b9c209b3ec240f2f56f83ad507b425ed096dfbbcacdc2558b6f47c13d91da413ab84f801ee49f61c"),
+					SignReceiver: types.Hex2Bytes("1172ed9142808860222ae220d99203a14ab6756ee65de9f091868905dafb195606239f0962886d54b029c08ee26551f057e97254ed3c78a728ac38bf1865ea1f1b"),
+				},
+			},
+		},
+		{
+			name: "invalid-state",
+			instance: &Instance{
+				selfID:      aliceID,
+				peerID:      bobID,
+				roleChannel: Sender,
+			},
+			args: args{
+				vpcState: VPCStateSigned{
+					VPCState: VPCState{
+						ID:              []byte("sample-id"),
+						Version:         big.NewInt(1),
+						BlockedSender:   big.NewInt(10),
+						BlockedReceiver: big.NewInt(20),
+					},
+				},
+			},
+		},
```

Comment:
What is a possible scenario for this to be true?

Reply:
This test case is to show that, whether the state is valid or not, it will be updated in the channel.

Reply:
new changes are reflecting that only valid states can be added